src/components/chatbox/utils.ts

```diff
@@ -0,0 +1,313 @@
+import { useEffect, useState, useRef } from 'react'
+
+// use a ref to check if an element's scrollbar is visible
+export function useScrollBarVisible() {
```

Comment:
💭 ⛏️ Very optional - the first three helpers & hooks in this file (`useScrollBBarVisible`, `mergeRefs`, `useThrottle`) seem both generic and useful enough that they might be worth lifting out to `src/lib` or something!